v1/microk8s-ha.yaml

```diff
@@ -0,0 +1,180 @@
+              systemctl disable apt-daily-upgrade.service apt-daily-upgrade.timer --now
+              systemctl disable unattended-upgrades.service --now
+              systemctl disable motd-news.service motd-news.timer --now
+              systemctl disable bluetooth.target --now
+              systemctl disable ua-timer.timer ua-timer.service --now
+              systemctl disable systemd-tmpfiles-clean.timer --now
+
+            - |
+              set -eux
+
+              # Setup microk8s
+              microk8s status --wait-ready
+
+              # The dns addon will restart the api server so you may see a blip in the availability
+              microk8s.enable dns
+              microk8s.kubectl rollout status deployments/coredns -n kube-system -w --timeout=600s
+
+              microk8s.enable rbac
+              microk8s.enable hostpath-storage
+              microk8s.kubectl rollout status deployments/hostpath-provisioner -n kube-system -w --timeout=600s
+
+              # MetalLB
+              IPADDR=$(ip -4 -j route get 2.2.2.2 | jq -r '.[] | .prefsrc')
+              microk8s enable metallb:$IPADDR-$IPADDR
+              microk8s.kubectl rollout status daemonset.apps/speaker -n metallb-system -w --timeout=600s
+
+              usermod -a -G snap_microk8s ubuntu
+
+        runcmd:
+        - DEBIAN_FRONTEND=noninteractive apt-get remove -y landscape-client landscape-common adwaita-icon-theme humanity-icon-theme
+        - DEBIAN_FRONTEND=noninteractive apt-get -y upgrade
+        - DEBIAN_FRONTEND=noninteractive apt-get -y autoremove
+
+        - |
+          # disable swap
+          sysctl -w vm.swappiness=0
+          echo "vm.swappiness = 0" | tee -a /etc/sysctl.conf
+          swapoff -a
+
+        - |
+          # disable unnecessary services
+          systemctl disable man-db.timer man-db.service --now
+          systemctl disable apport.service apport-autoreport.service  --now
+          systemctl disable apt-daily.service apt-daily.timer --now
+          systemctl disable apt-daily-upgrade.service apt-daily-upgrade.timer --now
+          systemctl disable unattended-upgrades.service --now
+          systemctl disable motd-news.service motd-news.timer --now
+          systemctl disable bluetooth.target --now
+          systemctl disable ua-timer.timer ua-timer.service --now
+          systemctl disable systemd-tmpfiles-clean.timer --now
+
+          # Disable IPv6
+          echo "net.ipv6.conf.all.disable_ipv6=1" | tee -a /etc/sysctl.conf
+          echo "net.ipv6.conf.default.disable_ipv6=1" | tee -a /etc/sysctl.conf
+          echo "net.ipv6.conf.lo.disable_ipv6=1" | tee -a /etc/sysctl.conf
+          sysctl -p
+
+        - |
+          # Set up cluster
+          # Strictly confined snap cannot see `/run`.
+          cp /run/microk8s-cloud-init.yaml ~ubuntu/
+          sudo -u ubuntu multipass launch 22.04 --name node-0 --memory 4G --cpus 2 --disk 30G --cloud-init ~ubuntu/microk8s-cloud-init.yaml
```

Comment:
@ktsakalozos recommends 4cpu4gb with 40GB disk space for each node, for general purpose initial use, so for the outer vm we'd probably need 14cpu16gb.